ztp/gitops-subscriptions/argocd/CoreReduction.md

```diff
@@ -0,0 +1,37 @@
+# Core Reduction with Workload Partitioning and Performance Profile
+
+Maximum number of cores can be allocated for customer workload only by pinning host and cluster workloads to a small number of cores. This is known as "core reduction" and be achieved by correctly configuring Workload Partitioning and Performance Profile.
+
+## Part 1 - Configure Performance Profile
+- Configure this using `PolicyGenTemplate`. See example [here](https://github.com/openshift-kni/cnf-features-deploy/blob/82ff3617a5e69b47b1f8d8b5d4a8db7719ab4bb4/ztp/gitops-subscriptions/argocd/example/policygentemplates/group-du-sno-ranGen.yaml#L99).
+- In this context we will look at [`reserved`](https://github.com/openshift-kni/cnf-features-deploy/blob/82ff3617a5e69b47b1f8d8b5d4a8db7719ab4bb4/ztp/gitops-subscriptions/argocd/example/policygentemplates/group-du-sno-ranGen.yaml#L105)
+  - The value here will CPU core numbers and are dependent on the hardware.
+  - This will pin all the host level services such as systemd and crio.
+- You can put the values as you see fit. But you can use "Performance Profile Creator" [tool](https://docs.openshift.com/container-platform/4.11/scalability_and_performance/cnf-create-performance-profiles.html#cnf-about-the-profile-creator-tool_cnf-create-performance-profiles) to select the suitable cores in `reserved` to maximize the benefit and ensure correctness (see WARNING). 
+  - look for `--reserved-cpu-count` when using Performance Profile Creator cli.
+- WARNING: PerformanceProfile's `reserved` and `isolated` must span ALL AVAILABLE CORES and not doing so will result in an undefined behaviour (generating it with tool above should take care of this behind-the-scenes)
+
+## Part 2 - Configure Workload Partitioning
+- Configure this using `SiteConfig`, See example [here](https://github.com/openshift-kni/cnf-features-deploy/blob/82ff3617a5e69b47b1f8d8b5d4a8db7719ab4bb4/ztp/gitops-subscriptions/argocd/example/siteconfig/example-sno.yaml#L59)
+- Set the value of `cpuset` the as the exact one used for `reserved` from `Part 1`
+
+## Part 3 - Apply configured SiteConfig and PolicyGenTemplate with GitOps
+The CRs are applied in reverse order. Note that PerformanceProfile is configured first (`Part 1`) for correctness and convenience, but it's expected to be applied as part of day-2 operations. To maximize core reduction benefits, both (SC and PGT) must be configured and applied eventually. 
+1. Apply SiteConfig
+2. Apply PolicyGenTemplate
+
+## Verification
+ssh into the node (e.g `oc debug node/myhost.rh.com`) and try: 
+- Look for cpu pinning with `taskset`
+  1. pinning done with PerformanceProfile and output must match `reserved`
+     - `pgrep systemd | while read i; do taskset -cp $i; done`
```

Comment:
What do you think of the idea of being more elaborate with the example here, something like this should allow a one command printout of the outside process checks, not the cleanest regex but just a thought.
```
pgrep "systemd|crio|kubelet" | while read i; do echo "CPUSet $(taskset -cp $i | grep -Po '[0-9]+[-,]+[0-9]+') for process $(ps -p $i -o comm=)"; done
```